File: src/CorporateMessengerWithAgile/Application/Dto/Summary/TaskItemSummaryDto.cs
namespace Application.Dto.Summary
{
    public class TaskItemSummaryDto : BaseDto
    {
        public Guid ProjectId { get; set; }
        public Guid AuthorId { get; set; }
        public Guid ResponsibleId { get; set; }
        public Guid? SprintWithLastMentionId { get; set; }
        public Guid? ParentTaskId { get; set; }
        public IReadOnlyList<Guid> SubtaskIds { get; set; } = null!;
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public int Priority { get; set; }
        public int Complexity { get; set; }
        public DateTime Deadline { get; set; }
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/TaskItemSummaryDto.ts
// src/models/entity/TaskItemSummaryDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const TaskItemSummaryDtoSchema = BaseDtoSchema.extend({
    projectId: GuidSchema,
    authorId: GuidSchema,
    responsibleId: GuidSchema,
    sprintWithLastMentionId: GuidSchema.nullish(),
    parentTaskId: GuidSchema.nullish(),
    subtaskIds: z.array(GuidSchema).optional(),
    title: z.string().min(1),
    description: z.string(),
    priority: z.number(),
    complexity: z.number(),
    deadline: z.coerce.date(),
});

export type TaskItemSummaryDto = z.infer<typeof TaskItemSummaryDtoSchema>;
